modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/delivery/catalog/resource/v1_0/test/CurrencyResourceTest.java

```diff
@@ -6,14 +6,228 @@
 package com.liferay.headless.commerce.delivery.catalog.resource.v1_0.test;
 
 import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.commerce.currency.model.CommerceCurrency;
+import com.liferay.commerce.currency.service.CommerceCurrencyLocalService;
+import com.liferay.commerce.product.model.CommerceChannel;
+import com.liferay.commerce.product.model.CommerceChannelRel;
+import com.liferay.commerce.product.service.CommerceChannelLocalService;
+import com.liferay.commerce.product.service.CommerceChannelRelLocalService;
+import com.liferay.commerce.test.util.CommerceTestUtil;
+import com.liferay.headless.commerce.core.util.LanguageUtils;
+import com.liferay.headless.commerce.delivery.catalog.client.dto.v1_0.Currency;
+import com.liferay.headless.commerce.delivery.catalog.client.pagination.Page;
+import com.liferay.headless.commerce.delivery.catalog.client.pagination.Pagination;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.test.rule.Inject;
 
-import org.junit.Ignore;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
 import org.junit.runner.RunWith;
 
 /**
  * @author Andrea Sbarra
+ * @author Michele Vigilante
  */
-@Ignore
 @RunWith(Arquillian.class)
 public class CurrencyResourceTest extends BaseCurrencyResourceTestCase {
+
+	@Before
+	@Override
+	public void setUp() throws Exception {
+		super.setUp();
+
+		_commerceChannel = CommerceTestUtil.addCommerceChannel(
+			testGroup.getGroupId(), RandomTestUtil.randomString());
+
+		User user = UserTestUtil.addUser(testCompany);
+
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
+			testCompany.getCompanyId(), testGroup.getGroupId(),
+			user.getUserId());
+
+		_addCurrency(_commerceChannel, randomCurrency());
+	}
+
+	@Override
+	@Test
+	public void testGetChannelByExternalReferenceCodeCurrenciesPage()
+		throws Exception {
+
+		super.testGetChannelByExternalReferenceCodeCurrenciesPage();
+
+		_testGetChannelByExternalReferenceCodeCurrenciesPageWithRestrictions();
+	}
+
+	@Override
+	protected Currency randomCurrency() throws Exception {
+		return new Currency() {
+			{
+				active = Boolean.TRUE;
+				code = StringUtil.toLowerCase(RandomTestUtil.randomString());
```

Comment:
Evaluate choosing an existing `code` for real test case scenarios. Also, they're usually uppercase.